dev-docs/RFCs/v4.0.0/polyfill-modes-rfc.md

```diff
@@ -12,10 +12,79 @@ Our current polyfill algorithm allocates cells to polygons based on whether the
 
 *Why is this important?*
 
+* Inclusion test
+
+A use case we want to support is testing whether a point could possibly be in a polygon by set containment. In Java this would look like:
+
+```java
+final int R = ...;
+// Indexed as res R
+Set<String> cellsCoveringPolygon = ...;
+
+public boolean polygonContainsPoint(double latitude, double longitude) {
+    String h3Index = h3.latLngToCell(latitude, longitude, R);
+    if (cellsCoveringPolygon.contains(h3Index)) {
+        // The polygon possibly contains the point, so use a more expensive point
+        // in polygon algorithm to determine.
+        return polygonContainsPointExpensive(latitude, longitude);
+    }
+    // The polygon definitely does not contain the point, the expensive check
+    // is not needed.
+    return false;
+}
+```
+
+* Analytics
+
+There is no convenient function to index a polygon into the H3 grid for analytics purposes. Polygons that are small enough, or shaped in specific ways, will not be converted into points, requiring workarouns in order to use them.
+
+* Spherical geometry consistency
+
+Most of the H3 library uses spherical geoemtry. For example, cell boundaries are spherical hexagons
+and pentagons. The `polyfill` function is different that it assumes Cartesian geometry. For
+consistency with the rest of the library, the `polyfill` functions should be able to use the same
+cell boundaries.
+
+* Very large polygons
+
+Polyfills of very large polygons requires allocating large blocks of memory, and spending large
+amounts of time in a library function without progress information being available to the caller.
+(To be determiend if this is in scope for this RFC or for another.)
+
 ## Approaches
 
 *What are the various options to address this issue?*
 
+On an API level, we will need to expose the containment mode and spherical/Cartesian choice as
```

Comment:
I think we might decide in v4 that polyfill is spherical, full stop, unless it's much slower this way. But I'm not sure there's a benefit to exposing both modes to the user, and it would mean a bunch of redundant code to check point inclusion and line intersection in both ways.

Reply:
I think it was raised offline that cartesian is useful when maps are drawn on top of web mercator maps.